### PR TITLE
Insert file path from file on clipboard

### PR DESCRIPTION
**Pull request by Garulf**
This PR allows Flow to retrieve the path of a file that has been copied from the file browser.

https://user-images.githubusercontent.com/535299/147052557-f0a0633d-c329-4076-9065-1fdb56f21ed2.mp4

### CODE REVIEW

File: Flow.Launcher/MainWindow.xaml.cs [hunk 1]
```diff
@@ -62,6 +67,30 @@ private async void OnClosing(object sender, CancelEventArgs e)
             Environment.Exit(0);
         }
 
+        private void onPreviewCanExecute(object sender, CanExecuteRoutedEventArgs e)
+        {
+            // Allow paste command to accept non-text items
```
Comment (jjw24) [inline Flow.Launcher/MainWindow.xaml.cs:72]: what does this mean sorry, pasting non-text into the query window?

**Comment by jjw24** [comment]
lets move the logic into MainViewModel

**Comment by jjw24** [comment]
is this ready?

**Comment by jjw24** [comment]
Just thinking, if you already have the file open in explorer, what usage scenario would you need to paste the file path into flow?

File: Flow.Launcher/MainWindow.xaml.cs (hunk 1 above)
Comment (Garulf) [reply Flow.Launcher/MainWindow.xaml.cs:85]: Yeah, normally TextBox elements ignore non-text data. With this we intercept all data input and capture it.

Comment (jjw24) [reply Flow.Launcher/MainWindow.xaml.cs:85]: oh so this is just required so you can retrieve path from the file object in clipboard?

**Comment by Garulf** [comment]
> Just thinking, if you already have the file open in explorer, what usage scenario would you need to paste the file path into flow?

It can help if you have a file but want to interact with it with Flow... kinda in reverse to what explorer plugin does.

**Comment by Garulf** [comment]
> lets move the logic into MainViewModel

I'll be honest I'm not sure how much can be moved into `MainViewModel`. I'll take a look.

**Comment by jjw24** [comment]
> > Just thinking, if you already have the file open in explorer, what usage scenario would you need to paste the file path into flow?
> 
> It can help if you have a file but want to interact with it with Flow... kinda in reverse to what explorer plugin does.

yeah but everything you can do in flow you can already do in windows explorer?

**Comment by Garulf** [comment]
For now, could be handy if I have a plugin that crops images or uploads them to a page.

**Comment by jjw24** [comment]
Sorry can you help me understand what those workflow would look like? I am still not sure how this could benefit when you already have windows explorer open already.

**Comment by Garulf** [comment]
> Sorry can you help me understand what those workflow would look like? I am still not sure how this could benefit when you already have windows explorer open already.

On second thought it's probably a bad example. Coming from MacOS it just always felt odd Wox or Flow don't know how to handle this.

**Comment by jjw24** [comment]
That's ok, could you give me an example of using this in flow, like how do you crop image then upload with this feature? i am lacking context because I don't use MacOS.

**Pull request closed without merging**